internal/http/services/appprovider/appprovider.go

```diff
@@ -189,39 +293,24 @@ func (s *svc) handleOpen(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func filterMimeTypes(mimeTypes map[string]*appregistry.AppProviderList) {
-	for m, providers := range mimeTypes {
+func filterAppsByUserAgent(mimeTypes []*appregistry.MimeTypeInfo, userAgent string) []*appregistry.MimeTypeInfo {
+	ua := ua.Parse(userAgent)
+	res := []*appregistry.MimeTypeInfo{}
+	for _, m := range mimeTypes {
 		apps := []*appregistry.ProviderInfo{}
-		for _, p := range providers.AppProviders {
+		for _, p := range m.AppProviders {
 			p.Address = "" // address is internal only and not needed in the client
-			// apps are called by name, so if it has no name you cannot call it. Therefore we should not advertise it.
-			if p.Name != "" {
+			// apps are called by name, so if it has no name it cannot be called and should not be advertised
+			// also filter Desktop-only apps if ua is not Desktop
+			if p.Name != "" && (ua.Desktop || !p.DesktopOnly) {
 				apps = append(apps, p)
 			}
 		}
 		if len(apps) > 0 {
-			mimeTypes[m] = &appregistry.AppProviderList{AppProviders: apps}
-		} else {
-			delete(mimeTypes, m)
+			res = append(res, m)
```

Comment:
We're adding all the apps for that mimetype, not using the filtered list from above.